src/gnu.rs

```diff
@@ -0,0 +1,109 @@
+use std::borrow::Cow;
+use std::cmp;
+use std::io::prelude::*;
+use std::io::{self, SeekFrom};
+use std::path::Path;
+
+use header::bytes2path;
+use other;
+use {Entry, Header};
+
+macro_rules! try_iter {
+    ($e:expr) => (match $e {
+        Ok(e) => e,
+        Err(e) => return Some(Err(e)),
+    })
+}
+
+/// dox
+pub struct GnuEntries<'a, R: 'a> {
+    inner: Box<Iterator<Item=io::Result<Entry<'a, R>>> + 'a>,
```

Comment:
Why are you using a boxed `Iterator` here? Can't you use a fully generic struct to avoid virtual dispatch?